Core/src/TrackFitting/GlobalChiSquareFitter.cpp

```diff
@@ -10,6 +10,30 @@
 
 #include "Acts/Definitions/TrackParametrization.hpp"
 
+void Acts::Experimental::updateGx2fParams(
+    BoundTrackParameters& params, const Eigen::VectorXd& deltaParamsExtended,
+    const std::size_t nMaterialSurfaces,
+    std::unordered_map<GeometryIdentifier, ScatteringProperties>& scatteringMap,
+    const std::vector<GeometryIdentifier>& geoIdVector) {
+  // update params
+  params.parameters() +=
+      deltaParamsExtended.topLeftCorner<eBoundSize, 1>().eval();
+
+  // update the scattering angles.
+  for (std::size_t matSurface = 0; matSurface < nMaterialSurfaces;
+       matSurface++) {
+    const std::size_t deltaPosition = eBoundSize + 2 * matSurface;
+    const GeometryIdentifier geoId = geoIdVector[matSurface];
+    const auto scatteringMapId = scatteringMap.find(geoId);
+    assert(scatteringMapId != scatteringMap.end() &&
+           "No scattering angles found for material surface.");
+    scatteringMapId->second.scatteringAngles().block<2, 1>(2, 0) +=
+        deltaParamsExtended.block<2, 1>(deltaPosition, 0).eval();
+  }
+
+  return;
```

Comment:
```suggestion
```
nitpick

Reply:
I was also thinking about it. Yesterday, I was more leaning towards no-return-void-functions, but then we already had the return in the other function below. I put it to stay consistent in this file. But I think, we should have a decision and brush over the whole repo :)